test_scripts/API/GetWayPoints/commonGetWayPoints.lua

```diff
@@ -0,0 +1,285 @@
+            EXPECT_HMICALL("BasicCommunication.SystemRequest")
+            :Do(function(_, d3)
+                self.hmiConnection:SendResponse(d3.id, "BasicCommunication.SystemRequest", "SUCCESS", { })
+                self.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate", { policyfile = policy_file_path .. "/" .. policy_file_name })
+              end)
+            mobileSession:ExpectResponse(corIdSystemRequest, { success = true, resultCode = "SUCCESS" })
+          end)
+        :Times(AtMost(1))
+      end
+    end)
+  os.remove(ptu_file_name)
+end
+
+function commonGetWayPoints.preconditions()
+  commonFunctions:SDLForceStop()
+  commonSteps:DeletePolicyTable()
+  commonSteps:DeleteLogsFiles()
+end
+
+--[[Module functions]]
+
+function commonGetWayPoints.activateApp(pAppId, self)
+  self, pAppId = commonGetWayPoints.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+  local mobSession = commonGetWayPoints.getMobileSession(self, pAppId)
+  local requestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+  commonTestCases:DelayedExp(commonGetWayPoints.minTimeout)
+end
+
+
+function commonGetWayPoints.getSelfAndParams(...)
+  local out = { }
+  local selfIdx = nil
+  for i,v in pairs({...}) do
+    if type(v) == "table" and v.isTest then
+      table.insert(out, v)
+      selfIdx = i
+      break
+    end
+  end
+  local idx = 2
+  for i = 1, table.maxn({...}) do
+    if i ~= selfIdx then
+      out[idx] = ({...})[i]
+      idx = idx + 1
+    end
+  end
+  return table.unpack(out, 1, table.maxn(out))
+end
+
+function commonGetWayPoints.getHMIAppId(pAppId)
+  if not pAppId then pAppId = 1 end
+  return hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+end
+
+function commonGetWayPoints.getMobileSession(self, pAppId)
+  if not pAppId then pAppId = 1 end
+  return self["mobileSession" .. pAppId]
+end
+
+function commonGetWayPoints.getMobileAppId(pAppId)
+  if not pAppId then pAppId = 1 end
+  return config["application" .. pAppId].registerAppInterfaceParams.appID
+end
+
+function commonGetWayPoints.getPathToSDL()
+  return config.pathToSDL
+end
+
+function commonGetWayPoints.postconditions()
+  StopSDL()
+end
+
+function commonGetWayPoints.raiPTU(ptu_update_func, self)
+  self, ptu_update_func = commonGetWayPoints.getSelfAndParams(ptu_update_func, self)
+  commonGetWayPoints.raiPTUn(1, ptu_update_func, self)
+end
+
+function commonGetWayPoints.raiPTUn(id, ptu_update_func, self)
+  self, id, ptu_update_func = commonGetWayPoints.getSelfAndParams(id, ptu_update_func, self)
```

Comment:
@LitvinenkoIra , suggestion is to combine raiPTU and raiPTUn to smth like registerAppWithPTU(id, func, self)